### ISSUE TITLE
package-nameing rule

### ISSUE THREAD

**Issue by Acollie** [mgechev/revive#1588]
**Is your feature request related to a problem? Please describe.**
While [Effective Go](https://go.dev/doc/effective_go#package-names) suggests that package names should be lower-case and single-word, it does not explicitly forbid other styles. In large projects or corporate environments, inconsistent naming (e.g., mixing snake_case, camelCase, or kebab-case) can lead to unreadable code usage and breaks internal style guides.

EG
```go
package bad_name

const(
    Limit = 10
)
```

```
bad_name.Limit
```
This forces the consumer to type bad_name.Limit, which may violate the organization's requirement for camelCase or simple lowercase.

**Describe the solution you'd like**
Name: `package-nameing` This rule as allows you to set a regex rule similar to [import-alias-naming](https://revive.run/RULES_DESCRIPTIONS.md#import-alias-naming) where it would take input in a regex pattern and this would allow for.
```toml
[rule.package-naming]
  arguments = ["^[a-z]+$"] # Enforces lowercase only, no underscores
```

**Describe alternatives you've considered**
I considered using [import-alias-naming](https://revive.run/RULES_DESCRIPTIONS.md#import-alias-naming). However, that rule only lints the usage (import side) and does not prevent the package definition itself from having a non-compliant name. Additionally, it does not help when imports are not aliased.

**Additional context**
I am happy to work on this change as it's very quick for me to implement.

**Comment by alexandear** [comment]
The [`var-naming`](https://revive.run/r#var-naming) rule handles package names. I think we can extend it to cover your case.

**Comment by Acollie** [comment]
> The [`var-naming`](https://revive.run/r#var-naming) rule handles package names. I think we can extend it to cover your case.

I did look [var-naming](https://revive.run/r#var-naming) that would work. I think personally var-naming doing quite a lot I prefer having smaller distinct rules. But I agree it could be a new parameter

**Comment by Acollie** [comment]
@alexandear I have a PR here https://github.com/mgechev/revive/pull/1592 for these changes with your suggestion :)

**Comment by alex-aizman** [comment]
disagree:

https://github.com/golangci/golangci-lint/issues/6232

`api.GetObject()` seems to be a good name - worked for us for years, as well as numerous other Go APIs that AIStore provides...

**Comment by alexandear** [comment]
I think it's better to do this is in a separate `package-naming` rule. 

`var-naming` is already overloaded with different checks.

**Comment by Acollie** [comment]
Should this be a follow up PR? If so I am happy to make this change
